storage/innobase/srv/srv0start.cc

```diff
@@ -1988,6 +1985,19 @@ void innodb_preshutdown()
 
   if (srv_read_only_mode)
     return;
+
+  #ifndef EMBEDDED_LIBRARY
+  if (srv_force_recovery < SRV_FORCE_NO_TRX_UNDO && srv_was_started &
+      !trx_rollback_is_active && trx_sys.any_active_transaction_recovered()) {
+      /* Trx rollback thread may not be started yet, since error happened before
+         innobase_tc_log_recovery_done(), thus it is started here to finish the
+         normal shutdown process.
+      */
+      trx_rollback_is_active = true;
```

Comment:
According to [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/-/CODING_STANDARDS.md) there should be no space before the `=`.